### PR TITLE
Features: Add TradeMarketAutoOpenChatFeature

### PR DESCRIPTION
**Pull request by kristofbolyai**
This feature opens trade market automatically if an input from user is required (filter, set amount, etc).

### CODE REVIEW

**Review by P0keDev — APPROVED** [review]

File: common/src/main/java/com/wynntils/features/user/TradeMarketAutoOpenChatFeature.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    public void onChatMessageReceive(ChatMessageReceivedEvent event) {
+        if (!WynnUtils.onWorld()) return;
+
+        if (TYPE_TO_CHAT_PATTERN.matcher(event.getCodedMessage()).matches()) {
+            openChatWhenContainerClosed = true;
+        }
+    }
+
+    @SubscribeEvent
+    public void onContainerClose(ScreenClosedEvent event) {
```
Comment (P0keDev) [inline common/src/main/java/com/wynntils/features/user/TradeMarketAutoOpenChatFeature.java:35]: do we use this same logic in legacy? just wondering if it's at all possible for the container to close before the message is sent

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/TradeMarketAutoOpenChatFeature.java:35]: I didn't check until I wrote this, but yes, same logic.

File: common/src/main/java/com/wynntils/features/user/TradeMarketAutoOpenChatFeature.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.regex.Pattern;
+import net.minecraft.client.gui.screens.ChatScreen;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class TradeMarketAutoOpenChatFeature extends UserFeature {
+
+    // Type the price in emeralds or type 'cancel' to cancel:
+    // Type the amount you wish to buy or type 'cancel' to cancel:
+    // Type the item name or type 'cancel' to cancel:
+    private static final Pattern TYPE_TO_CHAT_PATTERN = Pattern.compile("§6Type the .* or type 'cancel' to cancel:");
```
Comment (magicus) [inline common/src/main/java/com/wynntils/features/user/TradeMarketAutoOpenChatFeature.java:21]: In general, I like regexp patterns to include `^` and `$`, that makes sure we match the whole line, which makes the risk lower that we match on a partial message, potentially sent by a user.

File: common/src/main/java/com/wynntils/features/user/TradeMarketAutoOpenChatFeature.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    private boolean openChatWhenContainerClosed = false;
+
+    @SubscribeEvent
+    public void onChatMessageReceive(ChatMessageReceivedEvent event) {
+        if (!WynnUtils.onWorld()) return;
+
+        if (TYPE_TO_CHAT_PATTERN.matcher(event.getCodedMessage()).matches()) {
+            openChatWhenContainerClosed = true;
```
Comment (magicus) [inline common/src/main/java/com/wynntils/features/user/TradeMarketAutoOpenChatFeature.java:30]: Should we have some kind of additional safeguards here? Like that the chat message should arrive with a container open? Or that it is reset whenever a container is opened? I'm just (slightly) worried that this bool is going to get in an incorrect state, somehow.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/TradeMarketAutoOpenChatFeature.java:30]: Legacy impl is quite similar, I guess I could add a reset.

Comment (magicus) [reply common/src/main/java/com/wynntils/features/user/TradeMarketAutoOpenChatFeature.java:30]: It's up to you.

**Pull request merged**